### PR TITLE
fix(encoder): skip COLLECTIONS-PIPES override for scalar values (limitation #13)

### PR DESCRIPTION
**Pull request by robbyt**
Limitation #13 in the property-based fuzz: a scalar string value
under a COLLECTIONS-category key (`tags`, `items`, `list`, `array`,
`elements`, `values`, `data`) round-tripped with the wrapper bytes
attached to the key —

  {tags: "plain"} -> ||tags||plain|| -> {"||tags": "plain"}

The PIPES KV template `||${key}||${value}||` is byte-for-byte
identical to a single-item PIPES array's emission, so the lexer's
lazy/greedy `^(.+?)\|\|(.*)\|\|$` regex reattached the leading
`||` of the wrapper to the key. Lexer can't disambiguate the
shape; the fix is to stop the encoder from emitting it.

Fix: add `semanticStyle !== 'PIPES'` to the COLLECTIONS-override
guard in `selectSyntax`. When the override is skipped, scalar
strings fall through to the value-type branches and pick a
non-PIPES KV style (FAKE_COMMENT for odd lines, EQUALS for even
lines, etc.). All of those use a single key/value separator
distinct from `||`, so the round-trip is unambiguous.

Other semantic categories (PERSONAL → QUOTED, STATUS, TECHNICAL,
FINANCIAL, TEMPORAL) still apply — only PIPES is structurally an
array shape. Arrays under COLLECTIONS keys are unaffected because
array styling goes through `selectArrayStyle`, not `selectSyntax`.

Tests:

- New `src/__tests__/CollectionsKeyNonArrayValue.test.ts` (17
  cases): one per COLLECTIONS keyword as a scalar; edge values
  (`""`, `"||"`, `"a||b"`, "plain text with spaces"); mixed-case
  keys (override uses `lowerKey`); array-value no-regression for
  `tags`/`items`; other-category no-regression for `name`
  (PERSONAL → QUOTED) and `active` (STATUS).
- `src/__tests__/RoundTripFuzz.test.ts`: drop the (13) constraint
  branch, mark resolved in the docstring counter, remove the now-
  orphaned `COLLECTION_KEYS` constant. Pinned-seed fuzz still
  green.

422 tests pass, lint clean. CLI smoke for `{"tags":"plain"}` now
encodes as `//tags//plain` (FAKE_COMMENT) instead of the ambiguous
`||tags||plain||`.

BC break: yes (cosmetic). ROML produced before this PR for a
COLLECTIONS-key scalar value used PIPES KV; after this PR it uses
whatever non-PIPES style the parity-and-value-type branches pick.
This is a forward-only encoder change — the lexer still handles
every shape it did before, so older ROML continues to decode (the
broken way).

https://claude.ai/code/session_01XUQmsaUz2ieUbn5sGCdPRV

### CODE REVIEW

File: src/__tests__/RoundTripFuzz.test.ts [hunk 1]
```diff
@@ -220,12 +220,13 @@ describe('Round-trip property tests (fast-check)', () => {
  *     non-COLLECTIONS keys round-trip through the existing
```
[the remark this answers is not in the document]
Comment (robbyt) [reply src/__tests__/RoundTripFuzz.test.ts]: Right — I left the (12) note saying COLLECTIONS scalars were "constrained out under #13" but #13 is resolved in this same PR, so that sentence outed itself. Rewrote in `4c0bbaf`: scalar `|`-bearing strings under any key (COLLECTIONS or not, after this PR) round-trip via the value-type KV branches whose separators don't include `|`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/__tests__/RoundTripFuzz.test.ts [hunk 2]
```diff
@@ -360,11 +351,10 @@ function hasKnownLimitation(input: unknown): boolean {
     //      `|`-bearing values under COLLECTIONS keys are still
     //      constrained out by (13) below.
```
[the remark this answers is not in the document]
Comment (robbyt) [reply src/__tests__/RoundTripFuzz.test.ts]: Same staleness as the docstring case — the (12) inline comment also leaned on the `(13)` constraint that this PR removes. Rewrote in `4c0bbaf` to point at the value-type KV branches handling scalar `|`-bearing strings under any key (no longer needing the (13) branch as a backstop).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/__tests__/CollectionsKeyNonArrayValue.test.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    });
+  });
+
+  describe('regression: other semantic categories still apply their override', () => {
+    it('round-trips {name:"alice"} (PERSONAL → QUOTED)', () => {
+      const input = { name: 'alice' };
+      expect(roundTrip(input)).toEqual(input);
+    });
+
+    it('round-trips {active:"yes"} (STATUS)', () => {
```
[the remark this answers is not in the document]
Comment (robbyt) [reply src/__tests__/CollectionsKeyNonArrayValue.test.ts]: Good catch and a substantive one rather than just stale-doc — those tests claimed to verify the override "still applies" but the assertions only checked round-trip equality, which would hold whether the override fired or not (every semantic style produces a *different* shape, but each shape happens to round-trip cleanly, so equality alone wouldn't catch a regression).

Strengthened in `12d5c6e`: each PERSONAL/STATUS/TECHNICAL case now asserts against the emitted ROML — `name="alice"` for QUOTED, `active<` for BRACKETS, `&id&` for AMPERSAND. Also added a parallel "no longer routes scalars to PIPES KV" block that asserts `RomlFile.jsonToRoml({tags:"plain"})` does NOT match `^\|\|tags\|\|`, so the skip-PIPES override fix is positively pinned (the previous PIPES-KV emission was structurally distinguishable as `||tags||plain||`).

---
_Generated by [Claude Code](https://claude.ai/code)_
